target/linux/generic/pending-5.10/470-mtd-spi-nor-support-limiting-4K-sectors-support-base.patch

```diff
@@ -1,71 +0,0 @@
-From: Felix Fietkau <nbd@nbd.name>
-Date: Sat, 4 Nov 2017 07:40:23 +0100
-Subject: [PATCH] mtd: spi-nor: support limiting 4K sectors support based on
- flash size
-
-Some devices need 4K sectors to be able to deal with small flash chips.
-For instance, w25x05 is 64 KiB in size, and without 4K sectors, the
-entire chip is just one erase block.
-On bigger flash chip sizes, using 4K sectors can significantly slow down
-many operations, including using a writable filesystem. There are several
-platforms where it makes sense to use a single kernel on both kinds of
-devices.
-
-To support this properly, allow configuring an upper flash chip size
-limit for 4K sectors support.
-
-Signed-off-by: Felix Fietkau <nbd@nbd.name>
----
-
---- a/drivers/mtd/spi-nor/Kconfig
-+++ b/drivers/mtd/spi-nor/Kconfig
-@@ -24,6 +24,17 @@ config MTD_SPI_NOR_USE_4K_SECTORS
- 	  Please note that some tools/drivers/filesystems may not work with
- 	  4096 B erase size (e.g. UBIFS requires 15 KiB as a minimum).
- 
-+config MTD_SPI_NOR_USE_4K_SECTORS_LIMIT
-+	int "Maximum flash chip size to use 4K sectors on (in KiB)"
-+	depends on MTD_SPI_NOR_USE_4K_SECTORS
-+	default "4096"
-+	help
-+	  There are many flash chips that support 4K sectors, but are so large
-+	  that using them significantly slows down writing large amounts of
-+	  data or using a writable filesystem.
-+	  Any flash chip larger than the size specified in this option will
-+	  not use 4K sectors.
-+
- source "drivers/mtd/spi-nor/controllers/Kconfig"
- 
- endif # MTD_SPI_NOR
---- a/drivers/mtd/spi-nor/core.c
-+++ b/drivers/mtd/spi-nor/core.c
-@@ -2786,6 +2786,21 @@ static void spi_nor_info_init_params(str
- 	 */
- 	erase_mask = 0;
- 	i = 0;
-+#ifdef CONFIG_MTD_SPI_NOR_USE_4K_SECTORS
```

Comment:
I do not understand why this is not working based on the commit message.
If the flash has `SECT_4K` or `SECT_4K_PMC` set we add an erase operation for 4k in addition to the normal erase operation.

In `spi_nor_select_erase()` the call to `spi_nor_has_uniform_erase()` will return true and if `CONFIG_MTD_SPI_NOR_USE_4K_SECTORS` is set we would select the 4k erase operation and disable the other.

Reply:
> we would select the 4k erase operation and disable the other.

`CONFIG_MTD_SPI_NOR_USE_4K_SECTORS` already does this.

As is, the `LIMIT` does not work, and makes this 470 patch do nothing (only the depended on  `USE_4K_SECTORS` matters). 

For uniform erase regions devices:
`spi_nor_erase_sector` does not use the `erase_map`. It uses [`nor->erase_opcode`](https://elixir.bootlin.com/linux/v5.10.105/C/ident/erase_opcode), which was set in `spi_nor_select_erase`, called by `spi_nor_default_setup` near the start of `spi_nor_info_init_params`, before these 470 changes are run, or possibly in one of the [later `init_params` functions](https://elixir.bootlin.com/linux/v5.10.105/source/drivers/mtd/spi-nor/core.c#L2844).

Might fix it by preventing changing the [`wanted_size`](https://elixir.bootlin.com/linux/v5.10.105/source/drivers/mtd/spi-nor/core.c#L2577) if device size > LIMIT.